File: Baekjoon/브루트 포스/MinjuCho/1038.py
# 문제 풀이 링크: https://velog.io/@cuppizza/백준-1038-감소하는-수-C-브루트-포스
# 실행 시간: 96ms 메모리: 37976KB
from queue import deque
n = int(input())
q = deque()
for i in range(10):
    q.append(i)
cnt = 0
while cnt <= n:
    cnt += 1
    if len(q) == 0:
        print(-1)
        exit(0)
    num = q.popleft()
    for i in range(0, num % 10):
        q.append(num * 10 + i)
print(num)

File: Baekjoon/브루트 포스/MinjuCho/1038.cpp
// 문제 풀이 링크: https://velog.io/@cuppizza/백준-1038-감소하는-수-C-브루트-포스
// 실행 시간: 0ms 메모리: 2020KB
#include<iostream>
#include<queue>

using namespace std;
int main(void)
{
	long long n;
	cin >> n;
	queue<long long> q;
	for (int i = 0; i < 10; ++i)
	{
		q.push(i);
	}
	long long cnt = 0, num = -1;
	while (cnt <= n)
	{
		if (q.empty())
		{
			cout << "-1";
			return 0;
		}
		num = q.front();
		q.pop();
		++cnt;
		long long x = num % 10;
		for (long long i = 0; i < x; ++i)
		{
			q.push(num * 10 + i);
		}
	}
	cout << num;
}